tools/junitreport/junitreport.go

```diff
@@ -117,13 +120,17 @@ func main() {
 	arguments := flag.Args()
 	// If we are asked to summarize an XML file, that is all we do
 	if len(arguments) == 1 && arguments[0] == "summarize" {
-		summary, err := cmd.Summarize(input)
+		summary, err, ok := cmd.Summarize(input)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Error summarizing jUnit XML file: %v\n", err)
 			os.Exit(1)
 		}
 		fmt.Fprint(os.Stdout, summary)
-		os.Exit(0)
+		if ok {
+			os.Exit(0)
+		} else {
+			os.Exit(3)
```

Comment:
We *could* capture the output, `grep` through it to see if there number of failures is non-zero, but since this is *our* internal tool and only exists to serve this purpose, I didn't see an issue with defining its API like this.